libV2/schemaUtils.js

```diff
@@ -1659,15 +1665,51 @@ let QUERYPARAM = 'query',
       };
     }
 
-    if (requestContent[URLENCODED]) {
-      return resolveUrlEncodedRequestBodyForPostmanRequest(context, requestContent[URLENCODED]);
+    for (const contentType in requestContent) {
+      if (contentType === URLENCODED) {
+        encodedRequestBody = resolveUrlEncodedRequestBodyForPostmanRequest(context, requestContent[contentType]);
+        if (preferredRequestBodyType === 'first-listed') {
+          return encodedRequestBody;
+        }
+      }
+      else if (contentType === FORM_DATA) {
+        formDataRequestBody = resolveFormDataRequestBodyForPostmanRequest(context, requestContent[contentType]);
+        if (preferredRequestBodyType === 'first-listed') {
+          return formDataRequestBody;
+        }
+      }
+      else {
+        rawModeRequestBody = resolveRawModeRequestBodyForPostmanRequest(context, requestContent);
+        if (preferredRequestBodyType === 'first-listed') {
+          return rawModeRequestBody;
+        }
+      }
     }
 
-    if (requestContent[FORM_DATA]) {
-      return resolveFormDataRequestBodyForPostmanRequest(context, requestContent[FORM_DATA]);
+    // Check if preferredRequestBodyType is provided and return the corresponding request body if available
+    if (preferredRequestBodyType) {
+      if (preferredRequestBodyType === 'x-www-form-urlencoded' && encodedRequestBody) {
+        return encodedRequestBody;
+      }
+      else if (preferredRequestBodyType === 'form-data' && formDataRequestBody) {
+        return formDataRequestBody;
+      }
+      else if (preferredRequestBodyType === 'raw' && rawModeRequestBody) {
```

Comment:
We can remove `else if` and keep this `else` since it'll be default fallback.

Reply:
Updated, instead of ELSE I made it the final return statement.